apps/nextra/pages/en/build/cli/trying-things-on-chain/ledger.mdx

```diff
@@ -117,3 +117,266 @@ yes
+
+Open up a new terminal window to continue with the rest of the tutorial and leave the localnet running in the background.
+
+<Callout type="default">
+When you finally want to shut down the localnet, you can press `Ctrl+C` in the terminal window where you started the localnet.
+</Callout>
+
+### Set up localnet hot wallet profile
+
+In a fresh terminal window, set up a localnet profile for an account with a compromised private key:
+
+```sh filename="Terminal"
+aptos init --profile localnet-a
+```
+
+When prompted, choose `local` as the network, then enter the following private key:
+
+```sh filename="Terminal"
+0x44bbe12dd5e120d94f65fb3cdae1dc7017156a8f7da1ebfa876cabf48fa6f5e2
+```
+
+Once the profile is setup, store the account address for later:
+
+```sh filename="Terminal"
+export ACCOUNT_A=0x$(
+    aptos account lookup-address --profile localnet-a \
+    | jq -r '.Result'
+)
+echo $ACCOUNT_A
+```
+
+This should output:
+
+```sh filename="Terminal"
+0xaaaa29ce2bd42dec0bad87599d827c0bc565b2a83595489aa3dc3f5e43ebaaaa
+```
+
+<Callout type="default" emoji="🧠">
+For ease of identification this account has vanity address `0xaaaa...aaaa`, which was generated via [`optivanity`](https://github.com/econia-labs/optivanity).
+</Callout>
+
+### Inspect the account
+
+Inspect the account's `Account` resource:
+
+```sh filename="Terminal"
+aptos account list --profile localnet-a
+```
+
+Note that the authentication key is the same as the account address:
+
+```sh filename="Terminal"
+export AUTH_KEY_A=$(
+    aptos account list --profile localnet-a | jq -r \
+        '.Result[0]["0x1::account::Account"]["authentication_key"]'
+)
+echo $AUTH_KEY_A
+```
+
+This should also output:
+
+```sh filename="Terminal"
+0xaaaa29ce2bd42dec0bad87599d827c0bc565b2a83595489aa3dc3f5e43ebaaaa
+```
+
+### Set up localnet Ledger profile
+
+```sh filename="Terminal"
+aptos init --profile localnet-b --ledger
+```
+
+When prompted, choose `local` as the network, then derivation path index `0`.
+
+Once the profile is setup, store the account address for later:
+
+```sh filename="Terminal"
+export ACCOUNT_B=0x$(
+    aptos account lookup-address --profile localnet-b \
+    | jq -r '.Result'
+)
+echo $ACCOUNT_B
+```
+
+Check the authentication key for the account:
+
+```sh filename="Terminal"
+export AUTH_KEY_B=$(
+    aptos account list --profile localnet-b | jq -r \
+        '.Result[0]["0x1::account::Account"]["authentication_key"]'
+)
+echo $AUTH_KEY_B
+```
+
+### Rotate the authentication key of the hot wallet
+
+Rotate the authentication key of the hot wallet via [`aptos_framework::account::rotate_authentication_key_call`](https://github.com/aptos-labs/aptos-core/blob/415d7e12e94ebd833a910ed4138742f005149610/aptos-move/framework/aptos-framework/sources/account.move#L294):
+
+```sh filename="Terminal"
+aptos move run \
+    --args "hex:$AUTH_KEY_B" \
+    --profile localnet-a \
+    --function-id 0x1::account::rotate_authentication_key_call
+```
```

Comment:
Should probably just change the existing rotate key to use this then, and then you can get rid of most of this tutorial?

Reply:
Addressed as of 6b6cb8c, combined with https://github.com/aptos-labs/aptos-core/pull/11151